### PR TITLE
Contributor's guide

### LINKED ISSUE #950: Add contributor's guide
* [x] Instructions for adding a new page: https://github.com/duckdb/duckdb-web/blob/66272a975d16941225416c17c9e387753b026507/README.md#adding-a-new-page
* [x] Use U.S. English spelling
* [x] Point out that a large part of the documentation is auto-generated
* [x] Ensure that examples are self-contained is possible
* [x] Use relative URLs without the `.html` extension
* [x] Adhere some SQL guideline, e.g. use uppercase keywords (`SELECT`, etc.)
* [x] **Do not** use duplicate page headers, see #958
* [x] If your change is non-trivial, build the site locally before contributing
  * Use Docker if you have trouble setting up Jekyll #953
* [x] Mention that guides for third-party tools using DuckDB in general do not belong to the DuckDB documentation
  * These could be added to https://github.com/davidgasquez/awesome-duckdb

### PR DESCRIPTION
**Pull request by szarnyasg**
Fixes #950

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+**When creating a PR, please check the box to "Allow edits from maintainers".**
+
+## Formatting
+
+### Syntax
+
+* Use [GitHub's Markdown syntax](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) for formatting.
+* Please enclose code in blocks that are tagged with the appropriate language (e.g., \`\`\`sql CODE HERE \`\`\`). 
+* The title of the page should be encoded in the front matter's `title` property. The body of the pages should not repeat this title word-by-word.
+* Use **4 spaces** for indentation.
```
Comment (Mause) [inline CONTRIBUTING.md:37]: I'd prefer we use the backtick syntax for code blocks, I'm not even sure if you can specify a syntax using the indentation version

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+## Code of Conduct
+
+This project and everyone participating in it is governed by a [Code of Conduct](code_of_conduct.md). By participating, you are expected to uphold this code. Please report unacceptable behavior to [quack@duckdb.org](mailto:quack@duckdb.org).
+
+## Eligibility
+
+Before submitting a contribution, please check whether your contribution is eligible.
+
+1. Before creating a new page, please do a search for the proposed page in the [existing documentation](https://duckdb.org/docs/).
```
Comment (Mause) [inline CONTRIBUTING.md:11]: Maybe just direct people to use GitHubs search?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    * :x: `../../sql/statements/copy.html`
+    * :x: `/docs/sql/statements/copy`
+    * :x: `https://duckdb.org/docs/sql/statements/copy`
+* Referencing a specific section is possible using the label of the section:
+    * :white_check_mark: `../../sql/statements/copy#copy-from`
+
+## Achive and generated pages
+
+* The archive pages (e.g., <https://duckdb.org/docs/archive/0.5.1/>) contain documentation for old versions. Do not edit these pages.
+* Many of the documentation's pages are auto-generated. Before editing, please check the [`scripts/generate_docs.py`](scripts/generate_docs.py) script. Do not edit the generated content, instead, edit the source files (often found in the [`duckdb` repository](https://github.com/duckdb/duckdb)).
```
Comment (Mause) [inline CONTRIBUTING.md:67]: the generate_docs.py script should probably be renamed - it actually only generates the C API docs

People should look at `scripts/generate_all_docs.sh`instead

File: CONTRIBUTING.md (hunk 2 above)
Comment (carlopi) [reply CONTRIBUTING.md:11]: Even this: https://duckdb.org/docs/search

File: CONTRIBUTING.md (hunk 1 above)
Comment (szarnyasg) [reply CONTRIBUTING.md:37]: I mean indentation as in indentation in SQL code. Clarifying now.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+The addition of a new guide requires one additional edit:
+* Add a link to the new page within the Guides landing page: `docs/guides/index.md`
+
+**Please test your changes using the steps listed in the [Building](building.md) guide.**
+
+**When creating a PR, please check the box to "Allow edits from maintainers".**
+
+## Formatting
+
+### Syntax
```
Comment (Mause) [inline CONTRIBUTING.md:31]: Optional, but would be worth mentioning that quote blocks are styled specially to appear as info boxes

**Review by Mause — APPROVED** [review]

**Comment by Alex-Monahan** [comment]
Sorry for my prior comment, I see that you recommend the awesome list! Can we link to that somewhat prominantly if we are slimming down the IDE docs?

**Comment by szarnyasg** [comment]
Sure, see #975.

**Pull request merged**